### PR TITLE
Fix deletion of multiple files should show one confirmation dialog

### LINKED ISSUE #9473: Deleting multiple linked files should show a single confirmation dialog
Follow up to #8659

When multiple linked items are removed, JabRef will show the "Delete the selected file permanently from disk" dialogue for each item separately. E.g., if the user removes 3 linked items, the dialogue asking whether the respective file should be deleted from disk or should just be removed from entry will be shown 3 times.

This is not the expected behaviour - instead the user would expect to answer this dialogue once for all items that were selected. So if the user selects the option "Remove from entry" from the dialogue, then this should be applied to all linked entries that have been selected for removal (not for each entry separately).

*Originally posted by @AEgit*

To view the linked files of an entry, follow these steps:

1. Open JabRef and open the entry editor for an item to which multiple files are linked (e.g., article + several supplements)
2. Navigate to the "General" tab and then go to the row "File"

### PR DESCRIPTION
**Pull request by shreverr**
Fixes #9473 

Moved code of delete dialog from `LinkedFileViewModel` to `LinkedFilesEditorViewModel`.
Made a separate method in 'LinkedFilesEditorViewModel' for deleting multiple files `deleteFiles`.
Made a 'getDir' method in `LinkedFileViewModel` to get the path of file.

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
---
Deleting single File

image

Deleting Multiple Files

image

### CODE REVIEW

File: src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java [hunk 1]
```diff
@@ -376,26 +374,18 @@ public void moveToDefaultDirectoryAndRename() {
...
     /**
-     * Asks the user for confirmation that he really wants to the delete the file from disk (or just remove the link).
      *
      * @return true if the linked file should be removed afterwards from the entry (i.e because it was deleted
      * successfully, does not exist in the first place or the user choose to remove it)
      */
-    public boolean delete() {
-        Optional<Path> file = linkedFile.findIn(databaseContext, preferences.getFilePreferences());
+    public boolean delete(Optional<ButtonType> buttonType, ButtonType removeFromEntry, ButtonType deleteFromEntry) {
+        Optional<Path> file = getDir();
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java:382]: Please avoid passing Optionals as arguments.
Why at all is a ButtonType object being used in the ViewModel?
This does not seem right at all. The ViewModel should not have to deal with the ui elements, this breaks the mvvm pattern. Please use primitives here.

File: src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java [hunk 2]
```diff
@@ -589,4 +579,9 @@ private Supplier<BibEntry> wrapImporterToSupplier(Importer importer, Path filePa
             }
         };
     }
+
+    public Optional<Path> getDir() {
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java:583]: Please don't use abbreviations. Try to be more concise, what directory is being looked for?

**Review by calixtus — CHANGES_REQUESTED** [review]
Hi, thanks for your interes in JabRef development and your proposed changes. I took a quick look into it and am a bit concerned that the changes do not really work well as they are right now.
I believe the whole delete stuff should move to a new class extending SimpleCommand to make this undoable eventually. This command could then be called at any place and should not depend on JavaFX dialog buttons.

Thanks!

**Comment by Siedlerchr** [comment]
Closing this PR due to inactivity :zzz: 
Please reopen the PR if you plan to continue working on this

**Pull request closed without merging**